src/Orleans/Streams/PersistentStreams/PersistentStreamProvider.cs

```diff
@@ -60,11 +60,18 @@ public GrainFactoryServiceProvider(IStreamProviderRuntime providerRuntime)
             }
             public object GetService(Type serviceType)
             {
-                if (serviceType == typeof (GrainFactory))
+                var service = providerRuntime.ServiceProvider?.GetService(serviceType);
```

Comment:
and also if providerRuntime is null, getting GrainFactory, or any service from it should return null.